test/es-module/loader-add-hash.mjs

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable node-core/required-modules */
+import url from 'url';
+
+export default ({
+  resolve: parentResolve
+}) => ({
+  async resolve(specifier, parentModuleURL) {
+    const parentResolved = await parentResolve(specifier, parentModuleURL);
+    const request = new url.URL(parentResolved.url);
+    request.hash = '#hash';
+    return {
+      url: request.href,
+      format: parentResolved.format
+    };
+  }
+});
```

Comment:
Note we do have a folder called `test/fixtures/es-module-loaders` for the loaders themselves which may be worth using here for these new loader cases.

Reply:
I thought about using that but we also had other non-test esm files in this folder, can move if desired.

Reply:
Ideally we could refactor towards `test/es-module` just containing the direct top-level test files, with everything else in fixtures directories?

Reply:
👍 That can be done later in another PR